api/src/main/java/org/apache/iceberg/catalog/NamespaceIdentifier.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.catalog;
+
+/**
+ * Identifies a namespace in iceberg catalog
+ */
+public class NamespaceIdentifier {
+  private final String[] namespaceParts;
```

Comment:
What is `namespaceParts`? Is it composed of `schema` and `database`?  What are your thoughts of taking schema and database and having corresponding getters instead of  `namespaceParts`?

Reply:
We don't want Iceberg to force an implementation to use 2 levels or 3 levels, or require namespaces at all. That's why we support any number of levels as a namespace. In our infrastructure, we have catalogs and databases/schemas as a namespace, but other organizations may only use database.